lib/webauthn/attestation_statement/android_safetynet.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require "jwt"
+require "openssl"
+require "webauthn/attestation_statement/base"
+
+module WebAuthn
+  # Implements https://www.w3.org/TR/2018/CR-webauthn-20180807/#android-safetynet-attestation
+  module AttestationStatement
+    class AndroidSafetynet < Base
+      def self.default_trust_store
+        OpenSSL::X509::Store.new.tap { |trust_store| trust_store.set_default_paths }
+      end
+
+      def valid?(authenticator_data, client_data_hash, trust_store: self.class.default_trust_store)
+        trusted_attestation_certificate?(trust_store) &&
+          valid_signature? &&
+          valid_attestation_domain? &&
+          valid_version? &&
+          valid_nonce?(authenticator_data, client_data_hash) &&
+          cts_profile_match?
+      end
+
+      private
+
+      def trusted_attestation_certificate?(trust_store)
+        signing_certificates.each do |certificate|
+          trust_store.add_cert(certificate)
+        end
+        trust_store.verify(attestation_certificate)
+      end
+
+      def valid_signature?
+        JWT.decode(statement["response"], attestation_certificate.public_key, true, algorithm: 'RS256')
+      rescue JWT::VerificationError
+        false
+      end
+
+      def valid_attestation_domain?
+        subject = attestation_certificate.subject.to_a
+        subject.assoc('CN')[1] == "attest.android.com"
+      end
+
+      def valid_version?
+        !statement["ver"].empty?
```

Comment:
Couldn't find any documentation on what the format is supposed to be, so I went with a simple presence check.

Reply:
It's clear a clear ongoing question for people writing the standard. https://github.com/w3c/webauthn/issues/968

At most, I would add a FIXME or TODO comment here that point to that issue where the recommendation is being tracked.

Implementation-wise, I think this is the best we can do at this point 👍

Reply:
Added a TODO note :)